www/_blog/2022-06-30-error-handling.mdx

```diff
@@ -0,0 +1,407 @@
+---
+title: Error handling in Rust
+description: A complete guide to error handling in Rust
+author: ben
+tags: [rust, tutorial]
+thumb: ferris-error-handling.png
+cover: ferris-error-handling.png
+date: "2022-06-30T18:00:00"
+---
+
+No program is perfect, some make errors. This post goes over how to handle them.
+
+Before talking about Results and how to handle them we must first talk about panics:
+ 
+### Panics
+
+[Panics](https://doc.rust-lang.org/std/macro.panic.html) are exceptions a program can throw. It stops all execution in the current thread. When a panic is thrown it returns a short description of what went wrong as well as information about the position of the the panic.
+
+```rust
+fn main() {
+    panic!("error!");
+    println!("Never reached :(");
+}
+```
+
+Running the above causes:
+
+```
+thread 'main' panicked at 'error!', examples\panics.rs:2:5
+```
+
+They are similar to `throw` in JavaScript and other languages, in that the don't need a annotation on the function to run and they can pass through function boundaries. However in Rust, panics cannot be recovered from, there is no way to incept a panic in the current thread. 

```

Comment:
typos: 'in that the don't need a annotation' -> 'in that they don't need an annotation'?